### PR TITLE
*(both): bump golang to 1.18

### LINKED ISSUE #4920: Bump TiFlow Golang to 1.18
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #4920

### What is changed and how it works?

* bump golang dependencies to 1.18, such as go.mod, docker files
* upgrade linter.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by ben1009** [comment]
is this just upgrade go version or have other refactor ?

**Comment by 3AceShowHand** [comment]
> is this just upgrade go version or have other refactor ?

only go version, to make it compile, linter also upgraded, so some code changes.

File: dm/syncer/compactor.go [hunk 1]
```diff
@@ -202,14 +201,10 @@ func (c *compactor) compactJob(j *job) {
 		if prevJob.dml.Type() == sqlmodel.RowChangeDelete {
 			// DELETE + INSERT => INSERT with safemode
 			j.safeMode = true
-			skipReduce = true
+			j.dml.Reduce(prevJob.dml)
```
Comment (lance6716) [inline dm/syncer/compactor.go:204]: seems this logic is mistakenly reverted

File: go.mod [hunk 2]
```diff
@@ -98,10 +92,189 @@ require (
...
 	google.golang.org/grpc v1.43.0
-	gopkg.in/mgo.v2 v2.0.0-20190816093944-a6b53ec6cb22 // indirect
 	gopkg.in/yaml.v2 v2.4.0
 	upper.io/db.v3 v3.7.1+incompatible
 )
 
+require (
+	cloud.google.com/go v0.93.3 // indirect
+	cloud.google.com/go/storage v1.16.1 // indirect
+	github.com/99designs/keyring v1.1.5 // indirect
```
Comment (overvenus) [inline go.mod:102]: What are these? Can them be removed?

File: pkg/actor/system.go [hunk 3]
```diff
@@ -388,8 +388,7 @@ func (b *SystemBuilder) Build() (*System, *Router) {
 	router := NewRouter(b.name)
 	metricWorkingDurations := make([]prometheus.Counter, b.numWorker)
 	for i := range metricWorkingDurations {
-		metricWorkingDurations[i] =
-			workingDuration.WithLabelValues(b.name, strconv.Itoa(i))
+		metricWorkingDurations[i] = workingDuration.WithLabelValues(b.name, strconv.Itoa(i))
```
Comment (overvenus) [inline pkg/actor/system.go:391]: Is it reformatted by linters?

Comment (overvenus) [reply pkg/actor/system.go:391]: I see https://github.com/mvdan/gofumpt#added-rules

Comment (3AceShowHand) [reply pkg/actor/system.go:391]: yes, caused by linter, and sees rules cannot be disabled. - - `

File: go.mod (hunk 2 above)
Comment (3AceShowHand) [reply go.mod:102]: it was added by `go mod tidy`, seems cannot be removed

**Review by overvenus — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
Please complete your PR body and link to the issue.

**Review by lance6716 — APPROVED** [review]
lgtm for changes made on DM package

**Comment by Ehco1996** [comment]
@3AceShowHand  please upgrade tidb-tools dep to fix CI  see https://github.com/pingcap/tidb-tools/pull/605

**Pull request merged**